enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -2589,11 +2610,21 @@ class AdjointGenerator
         dsrc = Builder2.CreateIntToPtr(dsrc,
                                        Type::getInt8PtrTy(dsrc->getContext()));
 
-      auto call =
-          Builder2.CreateMemCpy(ddst, dstAlign, dsrc, srcAlign, new_size);
-      call->setAttributes(MTI.getAttributes());
-      call->setTailCallKind(MTI.getTailCallKind());
+      Value *memFunc = MTI.getCalledOperand();
+
+      auto rule = [&](Value *ddst, Value *dsrc) {
+        CallInst *call = dyn_cast<CallInst>(
+            Builder2.CreateCall(MTI.getFunctionType(), memFunc,
+                                {ddst, dsrc, new_size, isVolatile}));
+        call->setAttributes(MTI.getAttributes());
+        call->setMetadata(LLVMContext::MD_tbaa,
+                          MTI.getMetadata(LLVMContext::MD_tbaa));
```

Comment:
Also tbaa_struct (we use that as well, so this is needed if nesting AD).

Reply:
done